xamarin/Xamarin.Forms
# Set StatusBar color and StatusBarStyle for iOS Android and UWP from Page

### Description of Change ###
With the advent of the black theme on iOS and generally good opportunities for adjusting colors, there was a need to be able to change the color of the StatusBar on the page.

### Issues Resolved ### 


- fixes #7314

### API Changes ###
New properties for Page
``` cs
public Color StatusBarColor { get; set; } // color for StatusBar
public StatusBarStyle StatusBarStyle { get; set; } //style for StatusBar content (Dark or Light)
```

Added:
``` cs
Page.StatusBarColor;
Page.StatusBarStyle;
```
``` cs
public enum StatusBarStyle
{
      Default = 0,
      LightContent = 1,
      DarkContent = 2
}
```
### Important
For iOS it require changes into info.plist  (maybe it can be put into Xamarin.Forms project template for VS and VS for Mac)
If this flag is not set, then the StatusBarStyle will not be changed.
``` 
<key>UIViewControllerBasedStatusBarAppearance</key>
<false/>
```
The priority settings for the StatusBar are on the container pages TabbedPage, CaruselPage, NaviationPage. it is higher than just pages. And their StatusBar settings will overwrite the settings of the page.

### Platforms Affected ### 
- Core/XAML (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

@jsuarezruiz thanks for this animation
<img src="https://user-images.githubusercontent.com/6755973/68944097-52e99d80-07ad-11ea-9b13-577bfa8bc255.gif" width="400">

### Before/After Screenshots ### 
<img src="https://user-images.githubusercontent.com/4385716/67844969-53283e80-fb07-11e9-811c-c437f6dd7f12.png" width="300">  <img src="https://user-images.githubusercontent.com/4385716/68526623-b889e580-02e6-11ea-8bdf-a34e8695bf33.png" width="300">  
<img src="https://user-images.githubusercontent.com/4385716/68285274-a2321e80-0087-11ea-894a-0700f0064e3a.png" width="600">
### Testing Procedure ###
1. Open Controll Gallery
2. Page StatusBar Gallery
3. Move R, G, and B sliders
4. Toggle Switch
5. Press buttons to check different pages

### PR Checklist ###
